pkg/storage/unified/resource/search.go

```diff
@@ -452,6 +452,32 @@ func (s *searchSupport) GetStats(ctx context.Context, req *resourcepb.ResourceSt
 	return rsp, nil
 }
 
+func (s *searchSupport) RebuildIndexes(ctx context.Context, req *resourcepb.RebuildIndexesRequest) (*resourcepb.RebuildIndexesResponse, error) {
+	ctx, span := tracer.Start(ctx, "resource.searchSupport.RebuildIndexes")
+	defer span.End()
+
+	filterKeys := make([]NamespacedResource, 0, len(req.Keys))
+	for _, key := range req.Keys {
+		filterKeys = append(filterKeys, NamespacedResource{
+			Namespace: key.Namespace,
+			Group:     key.Group,
+			Resource:  key.Resource,
+		})
+	}
+
+	importTimes, err := s.getLastImportTimes(ctx, filterKeys)
+	if err != nil {
+		return &resourcepb.RebuildIndexesResponse{
+			Error: AsErrorResult(err),
+		}, nil
+	}
+
+	rebuiltCount := s.findIndexesToRebuild(importTimes, filterKeys, time.Now())
+	return &resourcepb.RebuildIndexesResponse{
```

Comment:
At this point, we're returning from gRPC call, but indexes are not rebuilt yet, only queued for rebuild. If we want to wait until rebuild has actually finished, we need to wait for all queued runs of `rebuildIndex`.

One way to do that could be:
* add a channel to `rebuildRequest`
* when `rebuildIndex` runs and there is a channel in `rebuildRequest`, it is closed at the end of `rebuildIndex`
* make `findIndexesToRebuild` include new channel in all `rebuildRequest`, and also return slice of these channels from `findIndexesToRebuild` itself.

Then in this method we can iterate over all channels and wait for all of them to be closed (or error out on context cancellation).